chia/wallet/wallet_state_manager.py

```diff
@@ -210,7 +211,10 @@ def sql_trace_callback(req: str) -> None:
         self.private_key = private_key
         self.main_wallet = await Wallet.create(self, main_wallet_info)
 
-        self.wallets = {main_wallet_info.id: self.main_wallet}
+        # mypy is saying:
+        # Dict entry 0 has incompatible type "uint32": "Wallet"; expected "uint32": "WalletProtocol"
+        # but Wallet implements WalletProtocol
+        self.wallets = {main_wallet_info.id: self.main_wallet}  # type: ignore[dict-item]
```

Comment:
That sounds like a complaint that `Wallet` does not satisfy `WalletProtocol`.  How was it verified that it does?  I'm trying to think through what else would trigger this.

Reply:
I added:

```python
        x: Wallet
        y: WalletProtocol = x
```

And uncommented the ignore here and got:

```
chia/wallet/wallet_state_manager.py:215: error: Incompatible types in assignment (expression has type "Wallet", variable has type "WalletProtocol")  [assignment]
chia/wallet/wallet_state_manager.py:215: note: "Wallet" is missing following "WalletProtocol" protocol member:
chia/wallet/wallet_state_manager.py:215: note:     coin_added
chia/wallet/wallet_state_manager.py:215: note: Following member(s) of "Wallet" have conflicts:
chia/wallet/wallet_state_manager.py:215: note:     Expected:
chia/wallet/wallet_state_manager.py:215: note:         def get_confirmed_balance(self, record_list: Optional[Set[WalletCoinRecord]] = ...) -> Coroutine[Any, Any, uint64]
chia/wallet/wallet_state_manager.py:215: note:     Got:
chia/wallet/wallet_state_manager.py:215: note:         def get_confirmed_balance(self, unspent_records: Any = ...) -> Coroutine[Any, Any, uint128]
chia/wallet/wallet_state_manager.py:215: note:     Expected:
chia/wallet/wallet_state_manager.py:215: note:         def get_max_send_amount(self, unspent_records: Optional[Set[WalletCoinRecord]] = ...) -> Coroutine[Any, Any, uint128]
chia/wallet/wallet_state_manager.py:215: note:     Got:
chia/wallet/wallet_state_manager.py:215: note:         def get_max_send_amount(self, records: Optional[Set[WalletCoinRecord]] = ...) -> Coroutine[Any, Any, int]
chia/wallet/wallet_state_manager.py:220: error: Dict entry 0 has incompatible type "uint32": "Wallet"; expected "uint32": "WalletProtocol"  [dict-item]
```

Reply:
(and yes, you shouldn't have to trick the protocol into telling you what is wrong...  there are options we can talk about.)

Reply:
oh, I was certain mypy would have told me if `Wallet` did not implement `WalletProtocol`..

Reply:
Hence my other question about having made sure all the proper complaints happened.  `:]`  I think it took me like three refactor PRs for the RPC hinting stuff before I got to the point that any of the RPC classes were actually checked.  Speaking of which, I ought to look back and see how far I got with that.